webapi/src/main/scala/org/knora/webapi/util/StringFormatter.scala

```diff
@@ -1500,4 +1502,23 @@ class StringFormatter private(knoraApiHttpBaseUrl: String) {
             case _ => false
         }
     }
+
+    /**
+      * Given the projectInfo calculates the project's data named graph.
+      *
+      * @param projectInfo the project's [[ProjectInfoV1]].
+      * @return the IRI of the project's data named graph.
+      */
+    def projectDataNamedGraph(projectInfo: ProjectInfoV1): IRI = {
+        if (projectInfo.shortcode.isDefined) {
+            OntologyConstants.NamedGraphs.DataNamedGraphStart + "/" + projectInfo.shortcode.get + "/" + projectInfo.shortname
+        } else {
+            OntologyConstants.NamedGraphs.DataNamedGraphStart + "/" + projectInfo.shortname
+        }
+    }
+
+    def isValidShortcode(shortcode: String): Boolean = {
+        // ToDo: do some checking
```

Comment:
Could you implement this TODO? There's already a regex sub-pattern for it, `ProjectIDPattern`, which you could use to make a regex for validating shortcodes.